x-pack/plugins/task_manager/server/kibana_discovery_service/kibana_discovery_service.ts

```diff
@@ -52,29 +54,31 @@ export class KibanaDiscoveryService {
   }
 
   private async scheduleUpsertCurrentNode() {
-    const lastSeenDate = new Date();
-    const lastSeen = lastSeenDate.toISOString();
-    try {
-      await this.upsertCurrentNode({ id: this.currentNode, lastSeen });
-      if (!this.started) {
-        this.logger.info('Kibana Discovery Service has been started');
-        this.started = true;
-      }
-    } catch (e) {
-      if (!this.started) {
-        this.logger.error(
-          `Kibana Discovery Service couldn't be started and will be retried in ${this.discoveryInterval}ms, error:${e.message}`
-        );
-      } else {
-        this.logger.error(
-          `Kibana Discovery Service couldn't update this node's last_seen timestamp. id: ${this.currentNode}, last_seen: ${lastSeen}, error:${e.message}`
+    if (!this.stopped) {
+      const lastSeenDate = new Date();
+      const lastSeen = lastSeenDate.toISOString();
+      try {
+        await this.upsertCurrentNode({ id: this.currentNode, lastSeen });
+        if (!this.started) {
+          this.logger.info('Kibana Discovery Service has been started');
+          this.started = true;
+        }
+      } catch (e) {
+        if (!this.started) {
+          this.logger.error(
+            `Kibana Discovery Service couldn't be started and will be retried in ${this.discoveryInterval}ms, error:${e.message}`
+          );
+        } else {
+          this.logger.error(
+            `Kibana Discovery Service couldn't update this node's last_seen timestamp. id: ${this.currentNode}, last_seen: ${lastSeen}, error:${e.message}`
+          );
+        }
+      } finally {
+        this.timer = setTimeout(
+          async () => await this.scheduleUpsertCurrentNode(),
+          this.discoveryInterval - (Date.now() - lastSeenDate.getTime())
```

Comment:
We should add a check to prevent this from being too small or negative. Seems like if it's been a while since the `lastSeenDate` has been updated, we could be using a very small value.

Reply:
Sorry I saw this in the issue and forgot to add it, resolved here [d066fb7](https://github.com/elastic/kibana/pull/193478/commits/d066fb76156df29bfba68cd822d70e64baffe728)